src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -458,3 +465,59 @@ killAccumulator acc xx statements
 
    | otherwise
    = return ((), s)
+
+-- | Simplify expressions with flattened primitives. Performs:
+--    * constant folding for common expressions and flat prims.
+--    * beta reduction
+--
+simpStatementExps :: Ord n => a -> Statement a n F.Prim -> Statement a n F.Prim
+simpStatementExps a_fresh statements
```

Comment:
might be able to use transformX here instead of transformUDStmt, to get rid of boilerplate. it doesn't really matter though